### ISSUE TITLE
PyTorch with CUDA 11 Support yields to DGL cudaErrorCudartUnloading Error

### ISSUE THREAD

**Issue by harshp8l** [dmlc/dgl#2271]
## 🐛 Bug

Building dgl from source upon a CUDA 11 container with cuda 11 compatible pytorch - yields with error:
```
dgl._ffi.base.DGLError: [10:15:47] /dgl/src/array/cuda/utils.cu:19: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading
```

## To Reproduce
Build cuda11 image using dockerfile mentioned in this PR:
https://github.com/aws/deep-learning-containers/pull/639

1. Clone DGL repo and checkout branch to version 0.5.2. Follow steps to build DGL from source
2. Run test: 
```
cd /dgl/examples/pytorch/gcn
python ./train.py --dataset cora --gpu 0
```

## Environment

 - DGL Version (e.g., 1.0):
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

## Additional context

CUDA Runtime and Driver Versions are correct and compatible since running ./deviceQuery results in:
```
deviceQuery, CUDA Driver = CUDART, CUDA Driver Version = 11.0, CUDA Runtime Version = 11.0, NumDevs = 8
Result = PASS
```
PyTorch was binary created with CUDA 11 compatibility; using the PyTorch nightly binaries also yields with the same issue mentioned above:
```
>>> import torch
>>> torch.cuda.is_available()
True
>>> torch.version.cuda
'11.0'
>>> torch.cuda.get_arch_list()
['sm_37', 'sm_50', 'sm_60', 'sm_70', 'sm_75', 'sm_80']
```

**Comment by BarclayII** [comment]
What is the complete error message (in particular the error code and message after what you showed here)?

**Comment by classicsong** [comment]
> What is the complete error message (in particular the error code and message after what you showed here)?

https://github.com/aws/deep-learning-containers/pull/639#issuecomment-706138503

**Comment by LILI-ZHOU** [comment]
I also met the same mistake, could you solve it

**Comment by yzh119** [comment]
@LILI-ZHOU, this issue should have been solved in #2308, could you please elaborate more on your environment and error message?

btw, the version of your CUDA runtime might be different from your CUDA driver, please make sure you installed DGL corresponding to your CUDA runtime version (by checking out `nvcc --version`).
